contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -2681,6 +2681,19 @@ rewriteBatchLevelStatement(
 				bool all_removed = removeTokenFromOptionList(expr, options, commas, ctx, getToken);
 				if (all_removed)
 					removeTokenStringFromQuery(expr, cctx->WITH(), ctx);
+
+				/* Check if execute_as_clause is present, and if is present check for caller clause */
+				bool execute_as_clause_check = false;
+				for (auto option : options)
+					if (option->execute_as_clause())
+						execute_as_clause_check = true;
+				if(execute_as_clause_check)
+				{
+					getToken = [](TSqlParser::Function_optionContext* o) { return o->execute_as_clause()->CALLER(); };
+					all_removed = removeTokenFromOptionList(expr, options, commas, ctx, getToken);
+					if (all_removed)
+						removeTokenStringFromQuery(expr, cctx->WITH(), ctx);
```

Comment:
We can remove following redundant code(after removing SCHEMABINDING and EXECUTE AS options, we can remove it after removing both options at the end.):
```
if (all_removed)
					removeTokenStringFromQuery(expr, cctx->WITH(), ctx);
```
instead of that after 
```
if(execute_as_clause_check)
{
...
}
<we can remove WITH() clause here>
```

Reply:
Fixed in next revision by replacing remove WITH() after removing both schemabinding and execute_As_clause.